### PR TITLE
formatter : add formatter module and its test

### PR DESCRIPTION
**Pull request by xavitator**
add formatter module and its test : #32

### CODE REVIEW

File: src/core/formatter.mli [hunk 1]
```diff
@@ -0,0 +1,12 @@
+(* 
+ * Formatter interface 
+ * DURAND - MARAIS © 2019
+ *)
+
+open Monad
+
+(* Recover string corresponding to the mustache template *)
```
Comment (maiste) [inline src/core/formatter.mli]: (\*\* needed

File: src/core/formatter.ml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+(* Get only variable declared in the template *)
+let get_template_var json =
+  match json with
+  | `Null | `Bool _ | `Float _ | `String _ | `A _ -> json
+  | `O l -> `O (List.filter (fun (str,_) -> str <> "template") l)
+
+(* Apply function corresponding to [List.map f l] where [f] is a [('a -> 'b choice)] function and [l] is a ['a list]. 
+ * Return a ['b list choice].
+*)
+let map_choice_list f l =
```
Comment (maiste) [inline src/core/formatter.ml]: It's about the choice monad : should be declared into Monad

File: src/core/formatter.ml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+           >>= (fun v -> Ok (str, v)))
+        with Not_found -> Ok(str, v)
+      in
+      map_choice_list format_object l 
+      >>= (fun l -> Ok (`O l))
+  in
+  aux templ user
+
+(* Take user json and template json, and return the header filled with user variable *)
+let formatter user templ =
```
Comment (maiste) [inline src/core/formatter.ml]: It could be nice to invert args according to format_user_var.

**Review by maiste — CHANGES_REQUESTED** [review]

File: src/core/formatter.ml (hunk 2 above)
Comment (xavitator) [reply src/core/formatter.ml]: What will be the name of the function?

File: src/core/formatter.mli (hunk 1 above)
Comment (xavitator) [reply src/core/formatter.mli]: I don't understand the comment.

File: src/core/formatter.ml (hunk 2 above)
Comment (maiste) [reply src/core/formatter.ml]: The same :smile:
It's just because it should be into Monad.

File: src/core/formatter.mli (hunk 1 above)
Comment (maiste) [reply src/core/formatter.mli]: Need a double \* instead of one to be accept by odoc

**Review by maiste — APPROVED** [review]

**Pull request merged**